test/e2e/cluster_extension_install_test.go

```diff
@@ -207,6 +207,7 @@ func TestClusterExtensionBlockInstallNonSuccessorVersion(t *testing.T) {
 		assert.Equal(ct, ocv1alpha1.ReasonSuccess, cond.Reason)
 		assert.Contains(ct, cond.Message, "resolved to")
 		assert.Equal(ct, &ocv1alpha1.BundleMetadata{Name: "prometheus-operator.1.0.0", Version: "1.0.0"}, clusterExtension.Status.ResolvedBundle)
+		assert.Equal(ct, &ocv1alpha1.BundleMetadata{Name: "prometheus-operator.1.0.0", Version: "1.0.0"}, clusterExtension.Status.InstalledBundle)
```

Comment:
If we don't actually check that this version has been installed, we may run into a scenario where we've resolved to `1.0.0` but haven't installed it yet. Then, if the user immediately changes the `ClusterExtension` to `1.2.0`, we're essentially installing `1.2.0` from scratch, and the non-successor limitation doesn't apply in that case. I don't personally think this is a bug.

This check is not necessary on `main` because when we finish `resolve` the `BundleDeployment` is immediately created with the installed version flags present, regardless of whether or not the `1.0.0` operator has finished installing yet. I think this is actually less desirable compared to the above scenario; for instance if the user quickly changed their mind about which version they wanted they wouldn't be able to install `1.2.0` even though `1.0.0` hasn't even installed yet.